group-coordinator/src/main/java/org/apache/kafka/coordinator/group/assignor/RangeAssignor.java

```diff
@@ -89,16 +93,17 @@ public MemberWithRemainingAssignments(String memberId, int remaining) {
     private Map<Uuid, List<String>> membersPerTopic(final AssignmentSpec assignmentSpec, final SubscribedTopicDescriber subscribedTopicDescriber) {
         Map<Uuid, List<String>> membersPerTopic = new HashMap<>();
         Map<String, AssignmentMemberSpec> membersData = assignmentSpec.members();
-
+        // Only add topic Ids to the map if they are present in the topic metadata.
         membersData.forEach((memberId, memberMetadata) -> {
             Collection<Uuid> topics = memberMetadata.subscribedTopicIds();
             for (Uuid topicId : topics) {
                 if (subscribedTopicDescriber.numPartitions(topicId) == -1) {
-                    throw new PartitionAssignorException("Member is subscribed to a non-existent topic");
+                    log.warn("Members are subscribed to topic " + topicId + " which doesn't exist in the topic metadata.");
```

Comment:
Shouldn't we keep what we had here?

Reply:
Yep I'll raise a different PR for range assignor changes